encode-and-decode-strings/flynn.cpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * For the number of given strings N, and the length of the longest string M,
+ * 
+ * Encode
+ * - Time complexity: O(N)
+ * - Space complexity: O(1)
+ * 
+ * Decode
+ * - Time complexity: O(NM)
+ * - Space complexity: O(M)
+ */
+
+class Codec {
+public:
+
+    // Encodes a list of strings to a single string.
+    string encode(vector<string>& strs) {
+        string res = "";
+        for (auto str : strs) {
+            res += to_string(str.size());
+            res.push_back('.');
+            res += str;
+        }
+        return res;
+    }
+
+    // Decodes a single string to a list of strings.
+    vector<string> decode(string s) {
+        vector<string> res;
+        int str_size = 0;
+        string tmp = "";
+
+        auto it = s.begin();
+        while (it != s.end()) {
+            do {
+                str_size = str_size * 10 + (*it - '0');
+                it++;
+            } while (*it != '.');
```

Comment:
`while`을 쓰다가 `do-while` 문을 써야 하는 특별한 Edge Case가 있을까 한참 생각을 해보았습니다. 별다른 이유가 없으시다면 그냥 `while` 문으로 통일하시는게 좀 더 코드의 가독성이 좋아지지 않을까요?

```suggestion
            while (*it != '.') {
                str_size = str_size * 10 + (*it - '0');
                it++;
            }
```

Reply:
`while`안에 `if(...) break;`를 넣고 싶지 않아서 그랬는데, `do...while`이 어색하게 느껴질 수도 있을 것 같습니다 
피드백 감사합니다, 다만 스타일의 문제인 것 같아서 이건 그냥 그대로 두겠습니다 ㅎㅎ